test/regression/regression.cc

```diff
@@ -476,7 +476,9 @@ int main(int argc, char **argv) {
     std::string ver(MODSECURITY_VERSION);
     std::string envvar("MODSECURITY=ModSecurity " + ver + " regression tests");
 
-    putenv(strdup(envvar.c_str()));
+    char *envvarptr = strdup(envvar.c_str());
+
+    putenv(envvarptr);
```

Comment:
Why do we need the strdup at all? The lifetime of envvar is main and it does not change, so we should be able to use envvar.c_str() directly

Reply:
> Why do we need the strdup at all?

I guess because `putenv()` might modify the string.

> The lifetime of envvar is main and it does not change, so we should be able to use envvar.c_str() directly

`envvar.c_str()` returns a `const char *` but `putenv()` does not allow it, only `char *`.

Never mind, see my newer commit [4085ff5](https://github.com/owasp-modsecurity/ModSecurity/pull/3114/commits/4085ff553624121d6cd76d20064394e4d6d21c1a), where I replaced `putenv()` by `setenv()`. Now SonarCloud has stopped complaining.